File: lib/domain/quest/quest_repository.dart
import 'package:allowance_questboard/domain/quest/quest.dart';
import 'package:allowance_questboard/domain/quest/quest_id.dart';

abstract interface class QuestRepository {
  Future<Quest?> find(QuestId questId);
  Future<List<Quest>?> findMembersBy(QuestId questId);
}

File: modules/edge/supabase/functions/quest/repository/quest_repository.ts
import { FamilyId } from "../../family/domain/value_object/family_id.ts";
import { Quest } from "../domain/quest.ts";
import { QuestId } from "../domain/value_object/quest_id.ts";

export class QuestRepository {
    findById(questId: QuestId): Promise<Quest | null> {
        throw new Error("Method findById() must be implemented.");
    }
    findByFamilyId(familyId: FamilyId): Promise<Quest | null> {
        throw new Error("Method findById() must be implemented.");
    }
}
